### DISCUSSION TITLE
Upgrade Guide for Custom Field/Tools/etc

### DISCUSSION THREAD

**Discussion by christian-heiko** [laravel/nova-issues#6630] [Q&A]
Im missing an Upgrade Guide on how to update custom Fields & Tools. 

Currently im doing the Trial & Error approach for every component, but would love a list of npm dependencies that can be dropped, API's & Components that have changed (like SelectControl) and just a general How to for upgrading custom components.

Or maybe somebody has already a list what plopped up during the Nova 5 upgrade.

**Comment by crynobone** [comment]
> API's & Components that have changed (like SelectControl) and just a general How to for upgrading custom components.

All Nova's Vue Component are not under any public API, if you used it then it 100% your responsibility to work on the compatibility. FYI, we make breaking changes to `SelectControl` in minor and patch releases.

> but would love a list of npm dependencies that can be dropped

Only require npm dependencies used directly by your components, other than that you can rely on the following doc and tool:

* https://nova.laravel.com/docs/v5/upgrade#updating-nova-components-custom-tool-cards-fields-filters
* https://github.com/laravel/nova-devtool

**Comment by VenoM9078** [comment]
I've been looking for this as well. Too many packages that are just unmaintained, you'd need to fork them and force upgrade them, even then - they are filled with issues caused by the Inertia upgrade.
